pkg/v1/cli/command/core/config.go

```diff
@@ -83,6 +83,11 @@ var setConfigCmd = &cobra.Command{
 		if len(args) > 2 {
 			return errors.Errorf("only path and value are allowed")
 		}
+
+		// Acquire tanzu config lock
+		config.AcquireTanzuConfigLock()
+		defer config.ReleaseTanzuConfigLock()
```

Comment:
Unfortunately, I don't think it is possible without refactoring how we are calling this functions because these external functions use `config.GetClientConfig` to get the config, update the config as part of the function and use `config.StoreClientConfig` to update the config. We would like to have the RW lock for this entire read/write operation.

I am not sure we can reliably add locking mechanism as part of `config.GetClientConfig` and `config.StoreClientConfig` individually.